File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Protection/protectunprotectdiagram.rb
module Asposediagramjava
  module ProtectUnprotectDiagram
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        diagram.getDocumentSettings().setProtectBkgnds(1)
        diagram.getDocumentSettings().setProtectMasters(1)
        diagram.getDocumentSettings().setProtectShapes(1)
        diagram.getDocumentSettings().setProtectStyles(1)

        # Save diagram
        diagram.save(data_dir + "ProtectUnprotectDiagram.vdx", Rjb::import('com.aspose.diagram.SaveFileFormat').VDX)

        puts "Applied protection on diagram successfully!"
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithProtection/ProtectUnprotectDiagram.php
<?php
namespace Aspose\Diagram\WorkingwithProtection;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;
class ProtectUnprotectDiagram{

    public static function run($dataDir=null){

        # Create instance of Diagram
        $diagram = new Diagram($dataDir . "drawing.vsd");

        $diagram->getDocumentSettings()->setProtectBkgnds(1);
        $diagram->getDocumentSettings()->setProtectMasters(1);
        $diagram->getDocumentSettings()->setProtectShapes(1);
        $diagram->getDocumentSettings()->setProtectStyles(1);

        # Save diagram
        $saveFileFormat=new SaveFileFormat();
        $diagram->save($dataDir . "ProtectUnprotectDiagram.vdx", $saveFileFormat->VDX);
        print "Applied protection on diagram successfully!".PHP_EOL;
    }
}